src/common/file_utils.cpp

```diff
@@ -125,5 +211,32 @@ std::vector<std::string> FileUtils::globFilePath(const std::string& path) {
     return result;
 }
 
+void FileUtils::truncateFileToSize(FileInfo* fileInfo, uint64_t size) {
+#if defined(_WIN32)
+    long offsetHigh = size >> 32;
```

Comment:
I think the size of long is dependent on the platform. Better to use `auto offsetHigh = (uint32_t)(size >> 32)`?

Reply:
It is platform-specific code though. I was following [the API for SetFilePointer](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-setfilepointer), which says the arguments are a `LONG`, which according to [this](https://learn.microsoft.com/en-us/windows/win32/winprog/windows-data-types) is just `typedef long LONG;`.

But I'll still change it; it's unintuitive.

Reply:
Though that actually would require casting the pointer back to a `long*`/`LONG*` when passing the argument to `SetFilePointer`, which seems a little redundant. Maybe it's better to just actually use `LONG` to indicate it's a microsoft-specified type for a microsoft-specified interface.